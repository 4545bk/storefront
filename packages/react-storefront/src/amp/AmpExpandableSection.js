/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React, { Component } from 'react'
import Typography from '@material-ui/core/Typography'
import withStyles from '@material-ui/core/styles/withStyles'
import ExpandLess from '@material-ui/icons/ExpandLess'
import ExpandMore from '@material-ui/icons/ExpandMore'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import AmpAccordion from './AmpAccordion'

export const styles = theme => ({
  section: {
    boxShadow: 'none',
    borderBottom: `1px solid ${theme.palette.divider}`,
    background: 'transparent'
  },
  title: {
    backgroundColor: 'transparent',
    padding: '12px 15px',
    borderStyle: 'none',
    outlineWidth: 0
  },
  toggle: {
    position: 'absolute',
    right: '18px',
    top: '13px'
  },
  expand: {
    display: 'block',
    'section[expanded] &': {
      display: 'none'
    }
  },
  collapse: {
    display: 'none',
    'section[expanded] &': {
      display: 'block'
    }
  },
  body: {
    backgroundColor: 'transparent',
    padding: `0 ${theme.margins.container}px`
  }
})

/**
 * An AMP-compatible expandable section based on amp-accordion.
 */
@withStyles(styles, { name: 'RSFAmpExpandableSection' })
export default class AmpExpandableSection extends Component {
  static propTypes = {
    /**
     * The title for the header of the expandable section
     */
    title: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),

    /**
     * Set to true to expand the panel
     */
    expanded: PropTypes.bool,

    /**
     * The icon to use for collapsed groups
     */
    ExpandIcon: PropTypes.func,

    /**
     * The icon to use for expanded groups
     */
    CollapseIcon: PropTypes.func,

    /**
     * Expanded section ID used by accordion
     */
    expandedSectionId: PropTypes.string
  }

  static defaultProps = {
    expanded: false,
    ExpandIcon: ExpandMore,
    CollapseIcon: ExpandLess
  }

  render() {
    let {
      classes,
      expanded,
      children = [],
      title,
      ExpandIcon,
      CollapseIcon,
      expandedSectionId
    } = this.props

    if (ExpandIcon === ExpandMore) {
      CollapseIcon = ExpandLess
    }

    const sectionAttributes = {}

    if (expanded) sectionAttributes.expanded = ''

    const section = (
      <section className={classes.section} {...sectionAttributes}>
        <Typography variant="subtitle1" component="h3" className={classes.title}>
          {title}
          <ExpandIcon className={classnames(classes.toggle, classes.expand)} />
          <CollapseIcon className={classnames(classes.toggle, classes.collapse)} />
        </Typography>
        <div className={classes.body}>{children}</div>
      </section>
    )

    if (expandedSectionId !== undefined) {
      return section
    }

    return <AmpAccordion>{section}</AmpAccordion>
  }
}
