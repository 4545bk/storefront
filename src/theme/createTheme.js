import { createMuiTheme, adaptV4Theme } from '@mui/material/styles';
import merge from 'lodash/merge'

/**
 * Creates the default theme for your React Storefront app.  See Material UI's theme documentation
 * for more info: https://mui.com/customization/default-theme/
 * @param {Object} values
 * @return {Object} A material UI theme
 */
export default function createTheme(values = {}) {
  const theme = createMuiTheme(
    adaptV4Theme(merge(
      {},
      {
        zIndex: {
          modal: 999,
          amp: {
            modal: 2147483646,
          },
        },
        headerHeight: 64,
        loadMaskOffsetTop: 64 + 56 + 4,
        drawerWidth: 330,
        overrides: {},
      },
      values,
    )),
  )

  if (!theme.maxWidth) {
    theme.maxWidth = `${theme.breakpoints.values.lg}px`
  }

  return theme
}
