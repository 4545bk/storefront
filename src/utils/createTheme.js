import { createMuiTheme } from '@material-ui/core/styles'
import merge from 'lodash/merge'

export default function createTheme(values = {}) {
  let config = {}

  merge(
    config,
    {
      zIndex: {
        modal: 999,
        amp: {
          modal: 2147483646,
        },
      },
      headerHeight: 64,
      loadMaskOffsetTop: 64 + 56 + 4,
      drawerWidth: 330,
      overrides: {},
    },
    values,
  )

  const theme = createMuiTheme(config)

  if (!theme.maxWidth) {
    theme.maxWidth = `${theme.breakpoints.values.lg}px`
  }

  return theme
}
