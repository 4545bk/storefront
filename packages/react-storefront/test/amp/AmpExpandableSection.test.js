/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React from 'react'
import { mount } from 'enzyme'
import AmpExpandableSection from '../../src/amp/AmpExpandableSection'
import createTheme from '../../src/createTheme'
import { MuiThemeProvider } from '@material-ui/core'

describe('AmpExpandableSection', () => {
  let theme

  beforeEach(() => {
    theme = createTheme()
  })

  it('should render', () => {
    expect(
      mount(
        <MuiThemeProvider theme={theme}>
          <AmpExpandableSection title="Title">Foo</AmpExpandableSection>
        </MuiThemeProvider>
      )
    ).toMatchSnapshot()
  })

  it('should render expanded', () => {
    expect(
      mount(
        <MuiThemeProvider theme={theme}>
          <AmpExpandableSection title="Title" expanded>Foo</AmpExpandableSection>
        </MuiThemeProvider>
      )
    ).toMatchSnapshot()
  })
})