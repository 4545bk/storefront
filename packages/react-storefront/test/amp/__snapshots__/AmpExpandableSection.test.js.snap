// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmpExpandableSection should render 1`] = `
<TestProvider>
  <Provider router={{...}} app={{...}} history={{...}} nextId={[Function: nextId]}>
    <MuiThemeProviderOld theme={{...}}>
      <WithStyles(AmpExpandableSection) title="Title">
        <AmpExpandableSection title="Title" classes={{...}} expanded={false} ExpandIcon={[Function: ShouldUpdate]} CollapseIcon={[Function: ShouldUpdate]}>
          <WithStyles(AmpAccordion)>
            <AmpAccordion classes={{...}}>
              <HelmetWrapper defer={true} encodeSpecialCharacters={true}>
                <SideEffect(NullComponent) defer={true} encodeSpecialCharacters={true} script={{...}}>
                  <NullComponent defer={true} encodeSpecialCharacters={true} script={{...}} />
                </SideEffect(NullComponent)>
              </HelmetWrapper>
              <amp-accordion expand-single-section={true} disable-session-states={true} class="RSFAmpAccordion-accordion-7">
                <section className="RSFAmpExpandableSection-section-1">
                  <WithStyles(Typography) variant="subtitle1" component="h3" className="RSFAmpExpandableSection-title-2">
                    <Typography variant="subtitle1" component="h3" className="RSFAmpExpandableSection-title-2" theme={{...}} classes={{...}} align="inherit" color="default" gutterBottom={false} headlineMapping={{...}} noWrap={false} paragraph={false}>
                      <h3 className="MuiTypography-root-8 MuiTypography-subtitle1-26 RSFAmpExpandableSection-title-2">
                        Title
                        <pure(ExpandMoreIcon) className="RSFAmpExpandableSection-toggle-3 RSFAmpExpandableSection-expand-4">
                          <ExpandMoreIcon className="RSFAmpExpandableSection-toggle-3 RSFAmpExpandableSection-expand-4">
                            <WithStyles(SvgIcon) className="RSFAmpExpandableSection-toggle-3 RSFAmpExpandableSection-expand-4">
                              <SvgIcon className="RSFAmpExpandableSection-toggle-3 RSFAmpExpandableSection-expand-4" classes={{...}} color="inherit" component="svg" fontSize="default" viewBox="0 0 24 24">
                                <svg className="MuiSvgIcon-root-43 RSFAmpExpandableSection-toggle-3 RSFAmpExpandableSection-expand-4" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true" role="presentation">
                                  <path d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z" />
                                  <path fill="none" d="M0 0h24v24H0z" />
                                </svg>
                              </SvgIcon>
                            </WithStyles(SvgIcon)>
                          </ExpandMoreIcon>
                        </pure(ExpandMoreIcon)>
                        <pure(ExpandLessIcon) className="RSFAmpExpandableSection-toggle-3 RSFAmpExpandableSection-collapse-5">
                          <ExpandLessIcon className="RSFAmpExpandableSection-toggle-3 RSFAmpExpandableSection-collapse-5">
                            <WithStyles(SvgIcon) className="RSFAmpExpandableSection-toggle-3 RSFAmpExpandableSection-collapse-5">
                              <SvgIcon className="RSFAmpExpandableSection-toggle-3 RSFAmpExpandableSection-collapse-5" classes={{...}} color="inherit" component="svg" fontSize="default" viewBox="0 0 24 24">
                                <svg className="MuiSvgIcon-root-43 RSFAmpExpandableSection-toggle-3 RSFAmpExpandableSection-collapse-5" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true" role="presentation">
                                  <path d="M12 8l-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14z" />
                                  <path fill="none" d="M0 0h24v24H0z" />
                                </svg>
                              </SvgIcon>
                            </WithStyles(SvgIcon)>
                          </ExpandLessIcon>
                        </pure(ExpandLessIcon)>
                      </h3>
                    </Typography>
                  </WithStyles(Typography)>
                  <div className="RSFAmpExpandableSection-body-6">
                    Foo
                  </div>
                </section>
              </amp-accordion>
            </AmpAccordion>
          </WithStyles(AmpAccordion)>
        </AmpExpandableSection>
      </WithStyles(AmpExpandableSection)>
    </MuiThemeProviderOld>
  </Provider>
</TestProvider>
`;

exports[`AmpExpandableSection should render expanded 1`] = `
<TestProvider>
  <Provider router={{...}} app={{...}} history={{...}} nextId={[Function: nextId]}>
    <MuiThemeProviderOld theme={{...}}>
      <WithStyles(AmpExpandableSection) title="Title" expanded={true}>
        <AmpExpandableSection title="Title" expanded={true} classes={{...}} ExpandIcon={[Function: ShouldUpdate]} CollapseIcon={[Function: ShouldUpdate]}>
          <WithStyles(AmpAccordion)>
            <AmpAccordion classes={{...}}>
              <HelmetWrapper defer={true} encodeSpecialCharacters={true}>
                <SideEffect(NullComponent) defer={true} encodeSpecialCharacters={true} script={{...}}>
                  <NullComponent defer={true} encodeSpecialCharacters={true} script={{...}} />
                </SideEffect(NullComponent)>
              </HelmetWrapper>
              <amp-accordion expand-single-section={true} disable-session-states={true} class="RSFAmpAccordion-accordion-58">
                <section className="RSFAmpExpandableSection-section-52" expanded="">
                  <WithStyles(Typography) variant="subtitle1" component="h3" className="RSFAmpExpandableSection-title-53">
                    <Typography variant="subtitle1" component="h3" className="RSFAmpExpandableSection-title-53" theme={{...}} classes={{...}} align="inherit" color="default" gutterBottom={false} headlineMapping={{...}} noWrap={false} paragraph={false}>
                      <h3 className="MuiTypography-root-59 MuiTypography-subtitle1-77 RSFAmpExpandableSection-title-53">
                        Title
                        <pure(ExpandMoreIcon) className="RSFAmpExpandableSection-toggle-54 RSFAmpExpandableSection-expand-55">
                          <ExpandMoreIcon className="RSFAmpExpandableSection-toggle-54 RSFAmpExpandableSection-expand-55">
                            <WithStyles(SvgIcon) className="RSFAmpExpandableSection-toggle-54 RSFAmpExpandableSection-expand-55">
                              <SvgIcon className="RSFAmpExpandableSection-toggle-54 RSFAmpExpandableSection-expand-55" classes={{...}} color="inherit" component="svg" fontSize="default" viewBox="0 0 24 24">
                                <svg className="MuiSvgIcon-root-94 RSFAmpExpandableSection-toggle-54 RSFAmpExpandableSection-expand-55" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true" role="presentation">
                                  <path d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z" />
                                  <path fill="none" d="M0 0h24v24H0z" />
                                </svg>
                              </SvgIcon>
                            </WithStyles(SvgIcon)>
                          </ExpandMoreIcon>
                        </pure(ExpandMoreIcon)>
                        <pure(ExpandLessIcon) className="RSFAmpExpandableSection-toggle-54 RSFAmpExpandableSection-collapse-56">
                          <ExpandLessIcon className="RSFAmpExpandableSection-toggle-54 RSFAmpExpandableSection-collapse-56">
                            <WithStyles(SvgIcon) className="RSFAmpExpandableSection-toggle-54 RSFAmpExpandableSection-collapse-56">
                              <SvgIcon className="RSFAmpExpandableSection-toggle-54 RSFAmpExpandableSection-collapse-56" classes={{...}} color="inherit" component="svg" fontSize="default" viewBox="0 0 24 24">
                                <svg className="MuiSvgIcon-root-94 RSFAmpExpandableSection-toggle-54 RSFAmpExpandableSection-collapse-56" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true" role="presentation">
                                  <path d="M12 8l-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14z" />
                                  <path fill="none" d="M0 0h24v24H0z" />
                                </svg>
                              </SvgIcon>
                            </WithStyles(SvgIcon)>
                          </ExpandLessIcon>
                        </pure(ExpandLessIcon)>
                      </h3>
                    </Typography>
                  </WithStyles(Typography)>
                  <div className="RSFAmpExpandableSection-body-57">
                    Foo
                  </div>
                </section>
              </amp-accordion>
            </AmpAccordion>
          </WithStyles(AmpAccordion)>
        </AmpExpandableSection>
      </WithStyles(AmpExpandableSection)>
    </MuiThemeProviderOld>
  </Provider>
</TestProvider>
`;
